<?xml version="1.0"?>
<launch>
  <!-- Always need this when playing back files-->
  <param name="/use_sim_time" value="true"/>

  <node pkg="rosbag" type="play" name="play_rosbag" args="$(find need4speed_scan_matching)/bags/levine_three_loops_plussome.bag --clock --delay 5" output="screen"/>
  
  <!-- Launch RVIZ -->
  <!-- Launch the racecar model -->
  <include file="$(find racecar_simulator)/launch/racecar_model.launch"/>
  <node pkg="rviz" type="rviz" name="rviz" args="-d $(find need4speed_scan_matching)/launch/scan_matching_visualization.rviz" output="screen"/>

  <!-- output="screen" allows us to see print statements-->
  <node pkg="need4speed_scan_matching" name="scan_matcher_node" type="scan_matcher" output="screen"/>
  <node pkg="need4speed_scan_matching" name="scan_matcher_quality_visualizer_node" type="scan_match_quality_visualizer" output="screen"/> 
  <node pkg="need4speed_scan_matching" name="scan_match_quality_node" type="scan_match_quality" output="screen">
    <rosparam command="load" file="$(find need4speed_scan_matching)/config/scan_matching_config.yaml"/>
  </node>

</launch>
